src/smart-components/Thread/context/ThreadProvider.tsx

```diff
@@ -73,6 +72,8 @@ export const ThreadProvider: React.FC<ThreadProviderProps> = (props: ThreadProvi
     renderUserProfile,
     onUserProfileMessage,
   } = props;
+  const propsMessage = props?.message;
+  const propsParentMessage = isThreadMessage(propsMessage) ? propsMessage?.parentMessage : propsMessage;
```

Comment:
is good to make a small function(not hook) and add a unit test,
or.. comment why we do this :)